app/models/team.rb

```diff
@@ -587,6 +595,30 @@ def get_feed(feed_id)
     self.feeds.where(id: feed_id.to_i).last
   end
 
+  # A newsletter header type is available only if there are WhatsApp templates for it
+  def available_newsletter_header_types
+    available = []
+    tbi = TeamBotInstallation.where(team_id: self.id, user_id: BotUser.smooch_user&.id.to_i).last
+    unless tbi.nil?
+      ['none', 'image', 'video', 'audio', 'link_preview'].each do |header_type|
+        mapped_header_type = {
+          'none' => 'none',
+          'image' => 'image',
+          'video' => 'video',
+          'audio' => 'video', # WhatsApp doesn't support audio header, so we convert it to video
+          'link_preview' => 'none'
+        }[header_type]
```

Comment:
You can move `mapped_header_type` array before `.each` and replace this block by `mapped_header_type[header_type]`